### PR TITLE
General Improvements

### PR DESCRIPTION
**Pull request by Akrog**
This PR includes several improvements:

- Return the output of the must-gather run into the files as `run.log`
- Add and document a debug mode
- Have per container logs
- Rename `<pod>-describe.log` to `<pod>-describe`

### CODE REVIEW

File: README.md
Comment (fmount) [inline README.md:49]: s/must-gather/openstack-must-gather

File: collection-scripts/gather
Comment (fmount) [inline collection-scripts/gather:21]: I see a few `shellcheck` related warnings, which might be good to analyze at this stage and then move forward. For instance here we have:

```

In collection-scripts/gather line 21:
/usr/bin/gather_run 2>&1 | tee $LOG
                               ^--^ SC2086 (info): Double quote to prevent globbing and word splitting.
```
that is just a matter of double quoting the variable.
Not a blocker though, I can address styling / cosmetic issues later (in a follow up) because there are a few

File: collection-scripts/gather
Comment (fmount) [inline collection-scripts/gather:24]: +1

File: collection-scripts/gather_ctlplane_resources
Comment (fmount) [inline collection-scripts/gather_ctlplane_resources:33]: awesome, it makes sense now that we have sidecar containers everywhere. I had this change in the backlog and I think it's cleaner having a log file for each container and get the "previous" log if there is one.

**Review by fmount — APPROVED** [review]
In general I'm +1 to this change, especially because there are interesting improvements in both the debug area, documentation and the way we get logs from pods (moving from a concatenation of all the containers to get a log per container, including the "previous" in case of failures).
There are a few shellcheck warnings but mostly related to the style.

**Comment by fmount** [comment]
I tested it locally and works as expected.
@Akrog feel free to go through the comments and we'll merge this as long as you're ready.
We still don't have CI and validations on this project, it's something that should be setup by the CI team soon.
For now the best chance we have is to run manual testing locally and make sure the resulting `openstack-must-gather` image is healthy.

File: README.md
Comment (Akrog) [reply README.md:49]: Done

File: collection-scripts/gather
Comment (Akrog) [reply collection-scripts/gather:21]: Done

File: collection-scripts/gather_ctlplane_resources
Comment (fmount) [inline collection-scripts/gather_ctlplane_resources:40]: -c "$c" here?

File: collection-scripts/gather_debug
Comment (fmount) [inline collection-scripts/gather_debug:8]: -e "${MSG}"

Comment (Akrog) [reply collection-scripts/gather_debug:8]: No need, echo command is ok like this as long as we don't start the MSG with a "-", which we won't do.

File: collection-scripts/gather_ctlplane_resources
Comment (Akrog) [reply collection-scripts/gather_ctlplane_resources:40]: No need, container names cannot have spaces, but I can change it to be consistent

File: collection-scripts/gather
Comment (fmount) [inline collection-scripts/gather:16]: one last question before merging this change.
what's the purpose of `OC_CALLED` ? I think I understand the meaning from the name but I'm not sure I got how it's used in the whole flow and which condition uses it (but it's probably my bad here

Comment (Akrog) [reply collection-scripts/gather:16]: This is leftover stuff I had from a previous version of the patch, in that one when OC_CALLED was false it would enter an endless loop.

I'll remove it now, thanks for catching this.

**Pull request merged**
